test/test_producer.py

```diff
@@ -23,21 +23,22 @@ def test_buffer_pool():
 
 
 @pytest.mark.skipif(not env_kafka_version(), reason="No KAFKA_VERSION set")
-@pytest.mark.parametrize("compression", [None, 'gzip', 'snappy', 'lz4'])
+@pytest.mark.parametrize("compression", [None, 'gzip', 'snappy', 'lz4', 'zstd'])
 def test_end_to_end(kafka_broker, compression):
-
     if compression == 'lz4':
-        # LZ4 requires 0.8.2
         if env_kafka_version() < (0, 8, 2):
-            return
-        # python-lz4 crashes on older versions of pypy
+            pytest.skip('LZ4 requires 0.8.2')
         elif platform.python_implementation() == 'PyPy':
-            return
+            pytest.skip('python-lz4 crashes on older versions of pypy')
+
+    if compression == 'zstd' and env_kafka_version() < (2, 1, 0):
+        pytest.skip('zstd requires kafka 2.1.0 or newer')
 
     connect_str = ':'.join([kafka_broker.host, str(kafka_broker.port)])
     producer = KafkaProducer(bootstrap_servers=connect_str,
                              retries=5,
                              max_block_ms=30000,
+                             api_version=env_kafka_version(),
```

Comment:
Why is this needed when it wasn't needed previously?

Reply:
auto discovered version would always come as 1.0.0 or similar. I think there is a PR now that actually deals with that  and updates the `conn.py::check_version` code, but i needed to force i higher version in order for the broker to not reject the requests

Reply:
Gotcha, that implies that for end users who are expecting this to just work, it won't work until the check version code is fixed. Hmm... I'd kinda rather see that merged first, I don't recall how much work that is, I'd have to check...

Reply:
fixed